components/browser/state/src/main/java/mozilla/components/browser/state/state/CustomTabSessionState.kt

```diff
@@ -52,12 +52,14 @@ fun createCustomTab(
     url: String,
     id: String = UUID.randomUUID().toString(),
     config: CustomTabConfig = CustomTabConfig(),
-    contextId: String? = null
+    contextId: String? = null,
+    engineState: EngineState = EngineState()
 ): CustomTabSessionState {
     return CustomTabSessionState(
         id = id,
         content = ContentState(url),
         config = config,
-        contextId = contextId
+        contextId = contextId,
+        engineState = engineState
```

Comment:
nit: The nice thing about `create[Custom]Tab()` is that you can use them as a "builder" without knowing about the internal state structure. Maybe we could add parameters for `EngineSession` (and if needed the `EngineSessionState`) and then assemble this internally (like we do for `ContentState`. Then a caller can assemble a tab without having to care about the `EngineState` wrapper. But it's also only a nit and we can do that later. :)

Reply:
Ah yeah, good point. That's inconsistent now. I will try and fix.

Reply:
OK, updated!